migrate.py

```diff
@@ -625,6 +646,258 @@ def copy_tests(plugin, coll, spec, args):
+                el[new_module_name] = el[key]
+                del el[key]
+        except LookupError:
+            pass
+
+
+def _rewrite_yaml_mapping_values(el, namespace, collection, spec):
+    for key, value in el.items():
+        if isinstance(value, Mapping):
+            _rewrite_yaml_mapping(el[key], namespace, collection, spec)
+        elif isinstance(value, list):
+            for idx, item in enumerate(value):
+                if isinstance(item, Mapping):
+                    _rewrite_yaml_mapping(el[key][idx], namespace, collection, spec)
+                else:
+                    if key == 'module_blacklist':
+                        if item in get_plugins_from_collection(collection, 'modules', spec):
+                            el[key][idx] = get_plugin_fqcn(namespace, collection, el[key][idx])
+
+                    el[key][idx] = _rewrite_yaml_lookup(el[key][idx], namespace, collection, spec)
+                    el[key][idx] = _rewrite_yaml_filter(el[key][idx], namespace, collection, spec)
+                    el[key][idx] = _rewrite_yaml_test(el[key][idx], namespace, collection, spec)
+        elif isinstance(value, str):
+            el[key] = _rewrite_yaml_lookup(el[key], namespace, collection, spec)
+            el[key] = _rewrite_yaml_filter(el[key], namespace, collection, spec)
+            el[key] = _rewrite_yaml_test(el[key], namespace, collection, spec)
+
+
+def _rewrite_yaml_lookup(value, namespace, collection, spec):
+    if not ('lookup(' in value or 'query(' in value or 'q(' in value):
+        return value
+
+    for plugin_name in get_plugins_from_collection(collection, 'lookup', spec):
+        if plugin_name in value:
+            value = value.replace(plugin_name, get_plugin_fqcn(namespace, collection, plugin_name))
+
+    return value
+
+
+def _rewrite_yaml_filter(value, namespace, collection, spec):
+    if '|' not in value:
+        return value
+
+    for filter_plugin_name in get_plugins_from_collection(collection, 'filter', spec):
+        imported_module = import_module('ansible.plugins.filter.' + filter_plugin_name)
+        fm = getattr(imported_module, 'FilterModule', None)
+        if fm is None:
+            continue
+        filters = fm().filters().keys()
+        for found_filter in [match[5] for match in FILTER_RE.findall(value)]:
+            if found_filter in filters:
+                value = value.replace(found_filter, get_plugin_fqcn(namespace, collection, found_filter))
+    return value
+
+
+def _rewrite_yaml_test(value, namespace, collection, spec):
+    if ' is ' not in value:
+        return value
+
+    for test_plugin_name in get_plugins_from_collection(collection, 'test', spec):
+        imported_module = import_module('ansible.plugins.test.' + test_plugin_name)
+        tm = getattr(imported_module, 'TestModule', None)
+        if tm is None:
+            continue
+        tests = tm().tests().keys()
+        for found_test in [match[5] for match in TEST_RE.findall(value)]:
```

Comment:
better avoid a temporary in-memory list creation
```suggestion
        for found_test in (match[5] for match in TEST_RE.findall(value)):
```